Documentation/troubleshooting.md

```diff
@@ -82,6 +82,24 @@ Note: The `ServiceMonitor` references a `Service` (not a `Deployment`, or a `Pod
 kubectl -n monitoring get secret prometheus-k8s -ojson | jq -r '.data["prometheus.yaml.gz"]' | base64 -d | gunzip | grep "my-service-monitor"
 ```
 
+### Debugging Why Resources Are Not Being Picked Up
+
+If you are encountering issues where PrometheusRules, PodMonitors, or ServiceMonitors are not being picked up by the Prometheus Operator please ensure that you have Prometheus operator >= v0.71.0. 
```

Comment:
I do believe that we need to expand the doc to cover all resources (pod/service monitors, probes, rules and scrapeconfigs) but for now we can keep it focused on ServiceMonitor.

I would also explain why that the operator may reject resources:

```
The Prometheus Operator will reject invalid resources and not reconcile them in the Prometheus configuration, for instance if the `ServiceMonitor object has a references to a secret which doesn't exist in the cluster. When it happens the Operator emits a Kubernetes event detailing the issue.

To check for events related to a `ServiceMontior`  object, you can use the following command:

...

{{< alert icon="👉" text="Prometheus Operator can emit Kubernetes events starting with v0.71.0."/>}}
```